### ISSUE TITLE
Ignore generated columns when export data

### ISSUE THREAD

**Issue by cdxf** [Wisser/Jailer#101]
### Is your feature request related to a problem? Please describe.
MYSQL can't import the exported file if the table contains generated columns:
`ERROR 3105 (HY000) at line 9402: The value specified for generated column 'column' in table 'table' is not allowed.`

https://dev.mysql.com/doc/refman/5.7/en/create-table-generated-columns.html
### Describe the solution you'd like
Ignore generated columns when export data or set the value to `DEFAULT`

> For [INSERT](https://dev.mysql.com/doc/refman/5.7/en/insert.html), [REPLACE](https://dev.mysql.com/doc/refman/5.7/en/replace.html), and [UPDATE](https://dev.mysql.com/doc/refman/5.7/en/update.html), if a generated column is inserted into, replaced, or updated explicitly, the only permitted value is DEFAULT.

**Comment by Wisser** [comment]
Generated columns are ignored when exporting if they are marked as "virtual" in the data model(editor):

![grafik](https://user-images.githubusercontent.com/861721/214515729-d870d2ec-e8fc-4169-a168-32307abc9ce2.png)

Virtual columns should be detected automatically. But this works only if the database driver is JDBC 4 compatible. Which driver do you use?

**Comment by cdxf** [comment]
It is not detected automatically I think
![image](https://user-images.githubusercontent.com/6419729/214516600-a6a91f6c-74d0-4342-8138-6eaa589db5f4.png)

I use mysql-connector-java-8.0.21

**Comment by Wisser** [comment]
That's strange. I just tested this again successfully with exactly this driver version.
Could you show me the DDL for this table (or at least that of the generated column)?

**Comment by cdxf** [comment]
DDL:

>  `is_hidden` tinyint(1) GENERATED ALWAYS AS expression_removed_for_brevity STORED`

Step to reproduce:
- New Connection (use remote db connection)
- Analyze Database
- Check the table details

**Comment by Wisser** [comment]
Thanks for the info.
I tested this again (also with the current MySQL driver) and found out the following:

a column:
`GENERATED ALWAYS AS ...`
is specified as "generated" by the driver.

A column:
`GENERATED ALWAYS AS ... STORED`
however not. (Which is a driver bug in my opinion).

I will try to find another way than via driver to find out in MySQL if a column is generated.
Currently the only solution to your problem is to set the "virtual" flag manually.

**Comment by Wisser** [comment]
An alternative way to get generated columns is to query the "INFORMATION_SCHEMA" tables.

I have adapted the configuration (based on version 14.1.1) for this purpose.

If you replace the file "jailer.xml" in your installation with the one in the attachment, your column should also be "virtual" after the analysis.

(Unpack the file for this purpose. GitHub does not allow XML files in the attachment. The destination folder is either ".jailer" in your user directory, or where you unzipped "jailer-n.zip". Depending on what you have installed) 

Please do a test on this.
Thanks in advance, 
Ralf

[jailer.xml.zip](https://github.com/Wisser/Jailer/files/10498126/jailer.xml.zip)

**Comment by Wisser** [comment]
Fixed in Release 14.2

**Comment by cdxf** [comment]
> An alternative way to get info about generated columns is to query the "INFORMATION_SCHEMA" tables.
> 
> I have adapted the configuration (based on version 14.1.1) for this purpose.
> 
> If you replace the file "jailer.xml" in your installation with the one in the attachment, your column should also be "virtual" after the analysis.
> 
> (Unpack the file for this purpose. GitHub does not allow XML files in the attachment. The destination folder is either ".jailer" in your user directory, or where you unzipped "jailer-n.zip". Depending on what you have installed)
> 
> Please do a test on this. Thanks in advance, Ralf
> 
> [jailer.xml.zip](https://github.com/Wisser/Jailer/files/10498126/jailer.xml.zip)

Sorry, I forget about this thread. Tried again and it works now in the latest version.
Thank for your work

**Comment by GrahamTheCoder** [comment]
In Jailer 15.4, using SQL2022 I find that by default it tries to specify values for the generated temporal columns. You can create a table that should repro the issue like this:
```sql

CREATE TABLE [dbo].[Person] (
    [Id] INT NOT NULL PRIMARY KEY IDENTITY,
    [RecordStartTime]         DATETIME2 (7) GENERATED ALWAYS AS ROW START HIDDEN NOT NULL,
    [RecordEndTime]           DATETIME2 (7) GENERATED ALWAYS AS ROW END HIDDEN   NOT NULL,
    PERIOD FOR SYSTEM_TIME ([RecordStartTime], [RecordEndTime])
)
WITH (SYSTEM_VERSIONING = ON (HISTORY_TABLE=[dbo].[PersonHistory], DATA_CONSISTENCY_CHECK=ON));
```

This query shows the column in case that helps:

```sql
SELECT *
FROM sys.columns
WHERE generated_always_type <> 0
AND object_id = OBJECT_ID('Person')
```

The workaround of manually marking virtual also worked fine.

Thanks for the great project!

**Comment by Wisser** [comment]
Thanks @GrahamTheCoder 
I have updated the request for the virtual columns accordingly.
You can find it here: https://sourceforge.net/projects/jailer/files/stage15.4.2/

Please try this out.

**Comment by GrahamTheCoder** [comment]
Yes that's working fine now, thanks again!

**Comment by Wisser** [comment]
Fixed in Release 15.5
